enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,485 @@
+
+
+
+### Goals
+
+- To provide a common experience across platforms for users and `openshift-install` developers
+- To be backwards compatible and fully satisfy the requirements of install-config type APIs.
+- To keep the user experience for day-zero operations unchanged or improved.
+- To not require any new runtime dependencies.
+- To maintain compatibility for hive, particularly regarding `destroy` functionality
+
+### Non-Goals / Future work
+
+- Non-goal: To maintain a consistent infrastructure footprint as clusters previously created with Terraform
+- Non-goal: Support openshift cluster creation by using any tools other than `openshift-install`, such as `clusterctl`
+- Future work: To optimize build processes or binary size (this will be prioritized ASAP in a separate design doc)
+- Future work: To use an existing management cluster to install OpenShift
+- Future work: To pivot the CAPI manifests to the newly-installed cluster to enable day-2 infrastructure management within the cluster.
+- Future work: Replace Machine API (MAPI) with CAPI for day-2 machine management. This enhancement assumes we are still using MAPI Day 2.
+- Future work: To provide an extensible framework to plug-in new infrastructure cloud providers.
+
+
+## Proposal
+
+The Installer will create CAPI infrastructure manifests based on user
+input from the install config; then, in order to provision cluster infrastructure,
+apply the manifests to CAPI controllers running on a local Kubernetes control-plane
+setup by [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest). 
+
+### Workflow Description
+
+**cluster creator** is a human user responsible for deploying a
+cluster. Note that the workflow does not change for this user.
+
+**openshift-install** is the Installer binary.
+
+1. The cluster creator provides an install-config and credentials
+2. (optional) The cluster creator runs `openshift-install create manifests`
+3. (optional) The cluster creator edits the newly created CAPI manifests.
+4. The cluster creator runs `openshift-install create cluster`
+5. `openshift-install` extracts binaries for kube-apiserver, etcd, CAPI infrastructure provider & cloud CAPI provider to the install dir
+6. `openshift-install` using `envtest` initializes a control plane locally on the Installer host
+7. `openshift-install` execs the CAPI infrastructure and cloud provider as subprocesses, pointing them to the local control plane
+8. `openshift-install` applies the CAPI manifests to the control plane
+9. The CAPI controllers provision cluster infrastructure based on the manifests
+10. `openshift-install` monitors the status of the CAPI resources as they are applied
+11. If the statuses are as expected, infrastrucutre has been provisioned and installation continues with the normal flow.
+
+In the case of an error in the final step, the Installer will bubble up resources with non-expected statuses.
```

Comment:
Is there anything in this workflow that is worth mentioning regarding creds management for the capi providers and how it differs from the tf workflow today?

Reply:
Even if it's the same, being explicit about that point would be good.

Reply:
Yes this is a great point. We should spell out the creds differences for each cloud platform.